Strategic/Queen Command.cpp

```diff
@@ -2716,60 +2716,54 @@ void BeginCaptureSquence( )
 
 void EndCaptureSequence( )
 {
-#ifdef JA2UB
-// no UB
-#else
-     
+#ifndef JA2UB
 	// Set flag...
 	if( !( gStrategicStatus.uiFlags & STRATEGIC_PLAYER_CAPTURED_FOR_RESCUE ) || !(gStrategicStatus.uiFlags & STRATEGIC_PLAYER_CAPTURED_FOR_ESCAPE) )
 	{
-		// CJC Dec 1 2002: fixing multiple captures:
-		//gStrategicStatus.uiFlags |= STRATEGIC_PLAYER_CAPTURED_FOR_RESCUE;
-
 		if ( gubQuest[ QUEST_HELD_IN_ALMA ] == QUESTNOTSTARTED )
 		{
-			// CJC Dec 1 2002: fixing multiple captures:
 			gStrategicStatus.uiFlags |= STRATEGIC_PLAYER_CAPTURED_FOR_RESCUE;
 			StartQuest( QUEST_HELD_IN_ALMA, gWorldSectorX, gWorldSectorY );
 		}
-		// CJC Dec 1 2002: fixing multiple captures:
-		//else if ( gubQuest[ QUEST_HELD_IN_ALMA ] == QUESTDONE )
-		else if (gubQuest[QUEST_HELD_IN_ALMA] != QUESTINPROGRESS && gubQuest[QUEST_HELD_IN_TIXA] == QUESTNOTSTARTED)
+		else if (gubQuest[QUEST_HELD_IN_TIXA] == QUESTNOTSTARTED)
 		{
-			// CJC Dec 1 2002: fixing multiple captures:
 			gStrategicStatus.uiFlags |= STRATEGIC_PLAYER_CAPTURED_FOR_RESCUE;
 			StartQuest(QUEST_HELD_IN_TIXA, gWorldSectorX, gWorldSectorY);
 		}
-		else if (gubQuest[QUEST_HELD_IN_ALMA] != QUESTINPROGRESS && gubQuest[QUEST_HELD_IN_TIXA] != QUESTINPROGRESS && gubQuest[QUEST_INTERROGATION] == QUESTNOTSTARTED)
+		else if (gubQuest[QUEST_INTERROGATION] == QUESTNOTSTARTED)
 		{
 			StartQuest( QUEST_INTERROGATION, gWorldSectorX, gWorldSectorY );
-			// CJC Dec 1 2002: fixing multiple captures:
 			gStrategicStatus.uiFlags |= STRATEGIC_PLAYER_CAPTURED_FOR_ESCAPE;
 
 			// OK! - Schedule Meanwhile now!
 			HandleInterrogationMeanwhileScene();
 		}
-		// CJC Dec 1 2002: fixing multiple captures
 		else
 		{
-			// !?!? set both flags
+			// Set both flags if we can't start any of the three POW quests
 			gStrategicStatus.uiFlags |= STRATEGIC_PLAYER_CAPTURED_FOR_RESCUE;
 			gStrategicStatus.uiFlags |= STRATEGIC_PLAYER_CAPTURED_FOR_ESCAPE;
 		}
 	}
 #endif
 }
 
-void EnemyCapturesPlayerSoldier( SOLDIERTYPE *pSoldier )
+int CalculateMaximumPrisonerAmount()
 {
-	UINT32					i;
-	BOOLEAN       fMadeCorpse;
-	INT32         iNumEnemiesInSector;
+	int maxPOWs = 0;
+#ifndef JA2UB
+	if (gubQuest[QUEST_HELD_IN_ALMA] == QUESTNOTSTARTED) { maxPOWs += std::size(gModSettings.iInitialPOWGridNo);  return maxPOWs; }
+	if (gubQuest[QUEST_HELD_IN_TIXA] == QUESTNOTSTARTED) { maxPOWs += std::size(gModSettings.iTixaPrisonPOWGridNo); return maxPOWs; }
+	if (gubQuest[QUEST_INTERROGATION] == QUESTNOTSTARTED) { maxPOWs += std::size(gModSettings.iMeanwhileInterrogatePOWGridNo); return maxPOWs; }

```

Comment:
is this supposed to short-circuit like that? If so, I'm guessing captured mercs can't go 30% to one facility, 50% to another and 20% to a third one?

Reply:
The quest gets reset if the player loses control of a prison sector, so it works correctly. 

I should clean this up some though, at first I was aiming on expanding possible POW amounts on prisons, and then knowing how many can currently fit in a prison would've been important. I altered it only slightly with the short circuits when I pivoted back to first getting POW feature simply working.